### PR TITLE
Upgrade bundle to be compatible with Sulu 2.0

### PR DESCRIPTION
**Pull request by niklasnatter**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| License | MIT

Requires: https://github.com/sulu/sulu/pull/4656

#### What's in this PR?

This PR refactors various parts of the codebase to make the bundle compatible with Sulu 2.0.

#### Why?

Because we want to be able to use this bundle in Sulu 2.0 projects 😉 

#### Todo:
- [x] Remove old javascript setup
- [x] Update composer dependencies to match sulu/sulu
- [x] Refactor Admin class
- [x] Refactor Extension class
- [x] Refactor translations to use json format
- [x] Add list configurations
- [x] Add form configurations
- [x] Remove unused routes
- [x] Add editorconfig
- [x] Install and configure php-cs-fixer
- [x] Enable PrettyCI for the project
- [x] Install and configure phpstan
- [x] Update travis configuration
- [x] Update functional test setup
- [x] Resolve `dispatch()` deprecations
- [x] Refactor controllers to use dependency injection
- [x] Use yml format for all routing files
- [x] Fix existing test cases
- [x] Fix existing phpstan errors
- [x] Adjust UPGRADE.md
- [x] Adjust README.md

### CODE REVIEW

File: Resources/doc/installation.md [hunk 1]
```diff
@@ -6,7 +6,13 @@ Install bundle over composer:
...
 ```
 
-Configure the routing:
+Add bundle to config/bundles.php`:
+
+```php
+    Sulu\Bundle\CommentBundle\SuluCommentBundle::class => ['all' => true],
+```
+
+Ad the routes of the bunlde to `config/routes/sulu_admin.yaml`:
```
Comment (alexander-schranz) [inline Resources/doc/installation.md]: typo

File: Admin/CommentAdmin.php [hunk 2]
```diff
@@ -50,25 +67,112 @@ public function __construct(SecurityCheckerInterface $securityChecker, $title)
...
+                ->setFormKey('comment_details')
+                ->setTabTitle('sulu_admin.details')
+                ->setEditRoute(static::COMMENT_EDIT_FORM_ROUTE)
+                ->addToolbarActions($formToolbarActions)
+                ->setParent(static::COMMENT_ADD_FORM_ROUTE)
+                ->getRoute(),
+            $this->routeBuilderFactory->createResourceTabRouteBuilder(static::COMMENT_EDIT_FORM_ROUTE, '/comments/:id')
+                ->setResourceKey('comments')
+                ->setBackRoute(static::COMMENT_LIST_ROUTE)
+                ->setTitleProperty('name')
```
Comment (alexander-schranz) [inline Admin/CommentAdmin.php]: 🤔 does a comment have a name?

File: Admin/CommentAdmin.php [hunk 3]
```diff
@@ -50,25 +67,112 @@ public function __construct(SecurityCheckerInterface $securityChecker, $title)
...
+                ->setFormKey('thread_details')
+                ->setTabTitle('sulu_admin.details')
+                ->setEditRoute(static::THREAD_EDIT_FORM_ROUTE)
+                ->addToolbarActions($formToolbarActions)
+                ->setParent(static::THREAD_ADD_FORM_ROUTE)
+                ->getRoute(),
+            $this->routeBuilderFactory->createResourceTabRouteBuilder(static::THREAD_EDIT_FORM_ROUTE, '/threads/:id')
+                ->setResourceKey('threads')
+                ->setBackRoute(static::THREAD_LIST_ROUTE)
+                ->setTitleProperty('name')
```
Comment (alexander-schranz) [inline Admin/CommentAdmin.php]: 🤔 does a thread have a name?

**Review by alexander-schranz — CHANGES_REQUESTED** [review]

File: Resources/config/doctrine/Thread.orm.xml [hunk 4]
```diff
@@ -16,8 +16,8 @@
             <generator strategy="AUTO"/>
         </id>
 
-        <field name="type" type="string" length="255"/>
-        <field name="entityId" type="string" column="entityId" length="255"/>
+        <field name="type" type="string" length="64"/>
```
Comment (wachterjohannes) [inline Resources/config/doctrine/Thread.orm.xml:19]: upgrade

**Comment by wachterjohannes** [comment]
@alexander-schranz this PR is now ready to review

File: Resources/config/lists/comments.xml [hunk 5]
```diff
@@ -0,0 +1,106 @@
+<?xml version="1.0" ?>
+<list xmlns="http://schemas.sulu.io/list-builder/list">
+    <key>comments</key>
+
+    <joins name="creator">
+            <join>
```
Comment (alexander-schranz) [inline Resources/config/lists/comments.xml]: indentation?

File: Resources/config/lists/comments.xml [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+                <entity-name>%sulu.model.user.class%</entity-name>
+                <field-name>%sulu.model.comment.class%.creator</field-name>
+            </join>
+            <join>
```
Comment (alexander-schranz) [inline Resources/config/lists/comments.xml]: I personally prefer have a empty line before a multiline opening tag for readability 🙈

File: Resources/config/lists/comments.xml [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+            <field-name>created</field-name>
+            <entity-name>%sulu.model.comment.class%</entity-name>
+        </property>
+
+        <property name="changed" translation="sulu_admin.changed" type="datetime">
+            <field-name>changed</field-name>
+            <entity-name>%sulu.model.comment.class%</entity-name>
+        </property>
+
+        <concatenation-property name="creator" visibility="always" glue=" "
```
Comment (alexander-schranz) [inline Resources/config/lists/comments.xml]: strange break (TODO we need a xml style linter 🤔 )

File: Resources/config/lists/threads.xml [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+            </field>
+            <field>
+                <field-name>lastName</field-name>
+                <entity-name>%sulu.model.contact.class%</entity-name>
+
+                <joins ref="creator"/>
+            </field>
+        </concatenation-property>
+
+        <concatenation-property name="changer" glue=" "
```
Comment (alexander-schranz) [inline Resources/config/lists/threads.xml]: strange break

File: Tests/Application/Kernel.php [hunk 9]
```diff
@@ -1,52 +1,38 @@
...
-            parent::registerBundles()
-        );
-
-        if (self::CONTEXT_ADMIN !== $this->getContext()) {
+        $bundles = parent::registerBundles();
+        $bundles[] = new SuluCommentBundle();
+
+        if (SuluTestKernel::CONTEXT_WEBSITE === $this->getContext()) {
             $bundles[] = new \FOS\RestBundle\FOSRestBundle();
+            $bundles[] = new SecurityBundle();
```
Comment (alexander-schranz) [inline Tests/Application/Kernel.php:30]: Do we really need this on the website context?
Is this documented or is it just needed for tests?

**Review by alexander-schranz — CHANGES_REQUESTED** [review]

File: Tests/Application/Kernel.php (hunk 9 above)
Comment (wachterjohannes) [reply Tests/Application/Kernel.php:30]: it is only neccesary to check if the user isset for a newly created comment

**Comment by wachterjohannes** [comment]
@alexander-schranz comments are fixed

**Pull request merged**